### DISCUSSION TITLE
Help implementing support for Eurom PAC Wifi Mobile Airco

### DISCUSSION THREAD

**Discussion by jriguera** [arendst/Tasmota#21653] [Support request] [answered]
During the last weeks I have been working on reverse engineering decoding the serial communication between the ESP and the MCU on a mobile airco unit Eurom PAC Wifi.  I have finished all work and published it at https://github.com/jriguera/eurom_pac_wifi_serial_potocol/blob/main/README.md

The way to control the airco is by transplanting the ESP with another one flashed with Tasmota. The operation is quite easy because there is a cable with a connector to/in the MCU box. There are no limitations of room (so, a lot of compatible Tasmota devices can be used) and everything can be reverted back to original setup by attaching the original ESP board (with the original firmware).

[The communication is serial at 9600 bits/s 8N1, with some extra headers and XOR checksums](https://github.com/jriguera/eurom_pac_wifi_serial_potocol/blob/main/protocol.md). I have the feeling the same protocol is being used in other devices of the same brand (mainly because the physical control panel is exactly the same in other devices, but I have no access to confirm this hypothesis).

I think the implementation should not be difficult for anybody familiar with Tasmota codebase: listening for RX datagrams, checking if the checksum is the same as the previous and if so, decoding status bytes and reporting. Unfortunately, right now, I do not have enough time to invest in this task, as it will take me some months to get familiar with the code and then start implementing it, for the time I will finish the summer will be gone.

I would be really grateful if somebody could program it, I can help testing it with my device and writing down documentation and manuals.

**Comment by sfromis** [comment, marked as answer]
If you use ESP32, you have the reasonably approachable Berry language with good support for dealing with a serial protocol. Probably much easier than diving into Tasmota native driver coding, and learning to know the code base.

**Reply by jriguera** [reply]
thanks for the idea! I did not had a ESP32 device at hand and I was not aware of the power of Berry!  I have a driver almost ready.

**Reply by jriguera** [reply]
I have implemented the driver in https://github.com/jriguera/tasmota_eurom_pac_wifi/tree/main/esp32.
Everything is working fine! 
Thanks!

**Comment by Pietjuhbel** [comment]
Being a Eurom device isn't the MCU just a Tuya MCU?
If so it should be quite easy to get sorted since Tasmota pretty solid Tuya support.

**Reply by jriguera** [reply]
As far as I know, Tuya uses a header `55aa00` which is not the case, this one is completely different. I also know that not all Eurom devices use Tuya.
